### ISSUE TITLE
Hebrew error with polyglossia 1.47 and lualatex

### ISSUE THREAD

**Issue by gildasius** [reutenauer/polyglossia#389]
When I use Hebrew with polyglossia 1.47 and LuaLaTeX on TeXLive 2019, I get an error message that I don't get with polyglossia 1.46. I got [some answers from tex.stack.exchange](https://tex.stackexchange.com/questions/527350/lualatex-and-polyglossia-1-47?noredirect=1#comment1333748_527350), but I thought I would also report it here as a possible bug. Here is my MWE:

%!TEX TS-program = lualatex
%!TEX encoding = UTF-8 Unicode

\documentclass{article}
\usepackage{fontspec}
\newfontfamily\hebrewfont{SBL Hebrew}[Script=Hebrew]
\usepackage{polyglossia}
\setdefaultlanguage[variant=us]{english}
\setotherlanguage{hebrew}

\begin{document}
Text in English and \texthebrew{עברית}. Text...
\end{document}

The messages I get

Package polyglossia Warning: Hebrew is not supported with LuaTeX.
(polyglossia)                I will proceed with the compilation, but
(polyglossia)                the output is not guaranteed to be correct
(polyglossia)                and may look very wrong. on input line 6.

(/usr/local/texlive/2019/texmf-dist/tex/lualatex/luabidi/luabidi.sty
(/usr/local/texlive/2019/texmf-dist/tex/lualatex/luabidi/luabidi-footnotes.def)
) (/usr/local/texlive/2019/texmf-dist/tex/latex/polyglossia/hebrewcal.sty
(/usr/local/texlive/2019/texmf-dist/tex/latex/polyglossia/cal-util.def))
Setting \if@calendar@hebrew
(/usr/local/texlive/2019/texmf-dist/tex/latex/polyglossia/babel-hebrewalph.def)

! Too many }'s.
l.120   \fi}
          %
! Extra \fi.
l.121 \fi

**Comment by moewew** [comment]
I can't say I have understood `\if...` matching, but if I replace

https://github.com/reutenauer/polyglossia/blob/6239583ea08b113c3851d5fb588649f0191d4198/tex/gloss-hebrew.ldf#L117-L119

with

```latex
\ifxetex
  \expandafter\@firstoftwo
\else
  \expandafter\@secondoftwo
\fi
  {\newcommand\xpg@hebrew@DigitsDotDashInterCharToks{\if@nonlatin\char"200E \fi}}
  {}
```

things appear to work again.

---

I think there might be spurious spaces from

https://github.com/reutenauer/polyglossia/blob/6239583ea08b113c3851d5fb588649f0191d4198/tex/gloss-hebrew.ldf#L123

and

https://github.com/reutenauer/polyglossia/blob/6239583ea08b113c3851d5fb588649f0191d4198/tex/gloss-hebrew.ldf#L130

with XeLaTeX.

**Comment by jspitz** [comment]
Thanks, @moewew. The problem here is that TeX stumbles over a `\fi` since it doesn't know an \if conditional (`\if@nonlatin` here), so `\ifxetex` is closed too early. See https://tex.stackexchange.com/questions/315749/

A solution (which I am going to take) is to use etoolbox's `\ifbool` rather than `\if`...`\fi` switches.

**Comment by jspitz** [comment]
This is fixed now. @reutenauer I am using the FIXED IN DEV label rather than closing the ticket. This way, other people who encounter the same issue might more easily find this ticket. I propose we close all FIXED IN DEV tickets after the release has been done.

**Comment by jspitz** [comment]
> I think there might be spurious spaces from
> 
> https://github.com/reutenauer/polyglossia/blob/6239583ea08b113c3851d5fb588649f0191d4198/tex/gloss-hebrew.ldf#L123
> 
> and
> 
> https://github.com/reutenauer/polyglossia/blob/6239583ea08b113c3851d5fb588649f0191d4198/tex/gloss-hebrew.ldf#L130
> 
> with XeLaTeX.

Thanks, should be addressed at e8e31f8fc5eb01f
